src/shady-css/common.js

```diff
@@ -0,0 +1,38 @@
+let matcher = {
```

Comment:
also, I'm frequently shocked at the ways VM optimizations fail, so it might be good to not nest the matchers and potentially cause a dereference